clang_delta/EmptyStructToInt.cpp

```diff
@@ -343,17 +437,120 @@ bool EmptyStructToInt::isValidRecordDecl(const RecordDecl *RD)
 
   const DeclContext *Ctx = dyn_cast<DeclContext>(CXXDef);
   TransAssert(Ctx && "Invalid DeclContext!");
+  int count = 0;
   for (DeclContext::decl_iterator I = Ctx->decls_begin(),
        E = Ctx->decls_end(); I != E; ++I) {
+    if ((*I)->isReferenced())
+        return false;
```

Comment:
On a second thought, this looks not correct. The reason is that implicity destructors can be referenced. For example, with the change, clang_delta failed to transform the code below:

    struct S { int f; };
    struct S s = {1};

Perhapse we could change the code to:

        if (!(*I)->isImplicit()) {
          if ((*I)->isReferenced())
            return false;
          ++count;
        }